src/mudlet.cpp

```diff
@@ -53,6 +53,7 @@
 #include "edbee/models/textgrammar.h"
 #include "edbee/texteditorwidget.h"
 #include "edbee/views/texttheme.h"
+#include "discord.h"
```

Comment:
This **has** to be conditional, at least by platform - it won't work on FreeBSD because the Desktop application is not available for that platform.  It may also not be wanted by some packagers - for instance since Discord is not packaged by Debian but provided by the Discord people themselves it is probable that @csmall will not be included to included it...

Based on my previous hackery, may I suggest an environmental variable `WITH_DISCORD` being set to `NO` to disable a `#define INCLUDE_DISCORD` on the supported platforms which is used to include/exclude the functionality.

Also you mentioned the code was MIT licenced - so it will need to be conditionally included in the "About dialog" - but it should be fairly straight-forward to conditionally add another licence to the listing there.

Reply:
I'll fix the include.

I don't want yet another flag for this, we already have a lot of flags for trivial things and I don't see players wanting to use a what would be a less than functional Mudlet

Reply:
Is it possible to load the functionality at runtime, depending on if the libraries are available?

Reply:
http://doc.qt.io/qt-5/qlibrary.html#details looks like so! I will investigate this approach.

Reply:
Conditional loading of the library dependent on availability will be a PITA for developers because they will have to arrange for the libraries concerned to both be available / not available to test both conditions won't they?  I guess adding/removing some symlinks is one crude way to do it.

The Discord libraries - are they C or C++ code?  I ask because that `QLibrary` seems to be aimed at [`dlopen`](https://en.wikipedia.org/wiki/Dynamic_loading)ìng (which I think is the name for run-time, explicit, loading of libraries on demand as opposed to the implicit, link stage of compilation, incorporation of library functions into an executable) C libraries only...

Reply:
All it takes is one `LD_PRELOAD_PATH` and only developers working with Discord code would need to do that - I prefer that option over more fobs in the build system.

It's C code, I think it'll be OK!